phoenix-core/src/main/java/org/apache/phoenix/query/ConnectionQueryServicesImpl.java

```diff
@@ -1588,15 +1596,19 @@ public GetVersionResponse call(MetaDataService instance)
                 }
             }
         }
-        if (isIncompatible) {
-            buf.setLength(buf.length()-1);
-            throw new SQLExceptionInfo.Builder(SQLExceptionCode.OUTDATED_JARS).setMessage(buf.toString()).build().buildException();
-        }
+
         if (systemCatalogTimestamp < MIN_SYSTEM_TABLE_TIMESTAMP) {
             throw new UpgradeRequiredException(systemCatalogTimestamp);
         }
     }
 
+    private String getClientandServerVersions(long serverJarVersion) {
```

Comment:
I guess this method can just be `getServerVersion` right? Since the client version is basically `MetaDataProtocol.CURRENT_CLIENT_VERSION`. That way, you're also not making it mandatory for this info to be returned as a string "Client version ... ".